tests/unit/test_hnsw_tiered.cpp

```diff
@@ -3090,6 +3095,192 @@ TYPED_TEST(HNSWTieredIndexTest, bufferLimitAsync) {
     EXPECT_EQ(tiered_index->indexLabelCount(), n_labels);
 }
 
+TYPED_TEST(HNSWTieredIndexTestBasic, RangeSearchCosine) {
```

Comment:
What is the purpose of these 3 tests? validating that cosine preprocessor is working as expected?

Reply:
Yes